app/feature/feature-addon-purchase/src/main/kotlin/com/hedvig/android/feature/addon/purchase/ui/customize/CustomizeTravelAddonDestination.kt

```diff
@@ -124,24 +130,29 @@ private fun CustomizeTravelAddonScreen(
   ) {
     when (val state = uiState) {
       is CustomizeTravelAddonState.Failure -> FailureScreen(state.errorMessage, reload, popBackStack)
-      CustomizeTravelAddonState.Loading -> HedvigFullScreenCenterAlignedProgress()
-      is CustomizeTravelAddonState.Success -> CustomizeTravelAddonScreenContent(
-        uiState = state,
-        navigateUp = navigateUp,
-        navigateToSummary = { quote ->
-          navigateToSummary(
-            SummaryParameters(
-              state.travelAddonOffer.title,
-              quote,
-              state.travelAddonOffer.activationDate,
-            ),
+      is CustomizeTravelAddonState.Loading -> {
+        HedvigFullScreenCenterAlignedProgress()
+      }
+      is CustomizeTravelAddonState.Success -> {
+        LaunchedEffect(state.summaryParamsToNavigateFurther) {
+          if (state.summaryParamsToNavigateFurther != null) {
+            navigateToSummary(state.summaryParamsToNavigateFurther)
+          }
+        }
+        if (state.summaryParamsToNavigateFurther == null) {
+          // todo: Stelios pls check here. This is bc I wanted to escape the brief flickering if we only use this to
+          //   load data and skip without showing the CustomizeTravelAddonScreenContent
```

Comment:
Yeah this looks reasonable to me, since we need to leave this screen immediately anyway.